src/index.js

```diff
@@ -62,15 +63,29 @@ const createRunner = ({ runtime = "worker_threads" } = {}) =>
           test => {
             onStart(test);
 
-            return this._pool
-              .run({ test, updateSnapshot, testNamePattern })
-              .then(
-                result => void onResult(test, result),
-                error => void onFailure(test, error),
-              );
+            return pool.run({ test, updateSnapshot, testNamePattern }).then(
+              result => void onResult(test, result),
+              error => void onFailure(test, error),
+            );
           },
           { concurrency: maxWorkers },
-        );
+        ).finally(async () => {
+          for (const { process } of pool.threads) {
+            // Use `process.disconnect()` instead of `process.kill()`, so we can collect coverage
+            // See https://github.com/nicolo-ribaudo/jest-light-runner/issues/90#issuecomment-2812473389
+            // Only override the first call https://github.com/tinylibs/tinypool/blob/dbf6d74282dd6031df8fc5c7706caef66b54070b/src/runtime/process-worker.ts#L61
+            const originalKill = process.kill;
+            process.kill = signal => {
+              if (!signal) {
+                process.disconnect();
+                process.kill = originalKill;
+              }
+              return originalKill.call(process, signal);
```

Comment:
Currently we call `originalKill.call(process, signal)` after `process.disconnect()`, is this expected?

Reply:
No, mistake.

I mean to only call `disconnect()` as your patch did to replace https://github.com/tinylibs/tinypool/blob/dbf6d74282dd6031df8fc5c7706caef66b54070b/src/runtime/process-worker.ts#L61

But it seems call `.disconnect()` before `.kill()` also works, let's keep it?

Reply:
No, I should return...